src/Extension/FluentVersionedExtension.php

```diff
@@ -325,6 +325,76 @@ public function existsInLocale($locale = null)
         return $this->isDraftedInLocale($locale) || $this->isPublishedInLocale($locale);
     }
 
+    /**
+     * Check if this record has modifications in this locale
+     * Fluent friendly version of @see Versioned::stagesDiffer()
+     *
+     * @param string|null $locale
+     * @return bool
+     */
+    public function stagesDifferInLocale($locale = null)
+    {
+        $record = $this->owner;
+        $id = $record->ID ?: $record->OldID;
+        $class = get_class($record);
+
+        // Need to check if it's versioned
+        if (!$record->hasExtension(Versioned::class)) {
+            return false;
+        }
+
+        // Need to check that it has stages and is not new
+        if (!$id || !$record->hasStages()) {
+            return false;
+        }
+
+        $locale = $locale ?: $this->getRecordLocale()->Locale;
```

Comment:
Please check that `getRecordLocale()` is not `null` before attempting to access a property on it.